mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -199,8 +238,8 @@ class LinearRegression(override val uid: String)
 
     val costFun = new LeastSquaresCostFun(instances, yStd, yMean, $(fitIntercept),
       $(standardization), featuresStd, featuresMean, effectiveL2RegParam)
-
-    val optimizer = if ($(elasticNetParam) == 0.0 || effectiveRegParam == 0.0) {
```

Comment:
Otherwise, there may be the case when `BreezeOWLQN` is used even if solver is set "l-bfgs".